presto-orc/src/test/java/com/facebook/presto/orc/TestReadBloomFilter.java

```diff
@@ -0,0 +1,143 @@
+
+public class TestReadBloomFilter
+{
+    @Test
+    public void test()
+            throws Exception
+    {
+        testType(TINYINT, ImmutableList.of(1L, 50L, 100L), 50L, 77L);
+        testType(SMALLINT, ImmutableList.of(1L, 5000L, 10_000L), 5000L, 7777L);
+        testType(INTEGER, ImmutableList.of(1L, 500_000L, 1_000_000L), 500_000L, 777_777L);
+        testType(BIGINT, ImmutableList.of(1L, 500_000L, 1_000_000L), 500_000L, 777_777L);
+        testType(DOUBLE, ImmutableList.of(1.11, 500_000.55, 1_000_000.99), 500_000.55, 777_777.77);
+
+        testType(VARCHAR, ImmutableList.of("a", "o", "z"), utf8Slice("o"), utf8Slice("w"));
+        testType(VARBINARY,
+                ImmutableList.of(new SqlVarbinary("a".getBytes(UTF_8)), new SqlVarbinary("o".getBytes(UTF_8)), new SqlVarbinary("z".getBytes(UTF_8))),
+                utf8Slice("o"),
+                utf8Slice("w"));
+        // Bloom filters are not supported for DECIMAL, FLOAT, DATE, TIMESTAMP, and CHAR
+    }
+
+    private static <T> void testType(Type type, List<T> uniqueValues, T inBloomFilter, T notInBloomFilter)
+            throws Exception
+    {
+        List<T> writeValues = newArrayList(limit(cycle(uniqueValues), 30_000));
+
+        try (TempFile tempFile = new TempFile()) {
+            writeOrcColumnHive(tempFile.getFile(), ORC_12, SNAPPY, type, writeValues);
+
+            // no predicate
+            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, type, Optional.empty(), true)) {
+                assertEquals(recordReader.nextBatch(), MAX_BATCH_SIZE);
+            }
+
+            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, type, Optional.empty(), false)) {
+                assertEquals(recordReader.nextBatch(), MAX_BATCH_SIZE);
+            }
+
+            // predicate for non-matching value
+            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, type, Optional.of(notInBloomFilter), true)) {
+                assertEquals(recordReader.nextBatch(), -1);
+            }
+
+            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, type, Optional.of(notInBloomFilter), false)) {
+                assertEquals(recordReader.nextBatch(), MAX_BATCH_SIZE);
+            }
+
+            // predicate for matching value
+            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, type, Optional.of(inBloomFilter), true)) {
+                assertEquals(recordReader.nextBatch(), MAX_BATCH_SIZE);
+            }
+
+            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, type, Optional.of(inBloomFilter), false)) {
+                assertEquals(recordReader.nextBatch(), MAX_BATCH_SIZE);
+            }
+        }
+    }
+
+    private static OrcBatchRecordReader createCustomOrcRecordReader(TempFile tempFile, OrcPredicate predicate, Type type, int initialBatchSize)
```

Comment:
@dilipkasana Dilip, I was doing a final pass and noticed that this method is not used. Could you remove it?